### ISSUE TITLE
Feature request: rename PowerShell Core to UltraShell

### ISSUE THREAD

**Issue by superbonaci** [PowerShell/PowerShell#16728]
### Summary of the new feature / enhancement

`PowerShell Core` should be renamed to `UltraShell`, otherwise it's mistaken with `Windows PowerShell`. Since this project is not an update of the other (with full retro-compatibility) they shouldn't be called the same. It's a sort of impersonation and I feel is misleading for the non techie user. In fact I know most people is not aware they are different programs.

**Comment by iSazonov** [comment]
> Since this project is not an update of the other (with full retro-compatibility) they shouldn't be called the same.

MSFT explicitly say that it is _next version_ of PowerShell - Windows PowerShell version is 5 and PowerShell version was 6 and now 7.
Keeping compatibility is main priority. 
No doubts PowerShell vNext will replace  Windows PowerShell 5.1 in future.

**Comment by superbonaci** [comment]
I've opened several bugs due to script incompatibility between version 5 and 7 (#12446 #12443). Keeping compatibility "priority" is not any guarantee of any kind. It's like `python2` and `python3`.

**Comment by jborean93** [comment]
Those 2 bugs you mention aren't really related to Ansible. #12446 is about a native command not being found on the system with isn't something PowerShell provides you. #12443 is about a custom script that isn't shipped at all by PowerShell but some 3rd party. That's not to say there aren't any incompatibility issues between the 2 versions but that's mostly down to the changes .NET Core (now just .NET) made to make it cross platform.

**Comment by jhoneill** [comment]
The names are "PowerShell" and "Windows PowerShell". The executables have different names pwsh  vs PowerShell. 

Neither of the examples you quoted are incompatibilities. - both are user errors an .EXE wasn't visible, and a script hadn't been installed (to the right place at least).
 
Genuine incompatibilities do exist, mostly with modules which require an earlier version of .NET - which has retained its name.

**Comment by superbonaci** [comment]
> Neither of the examples you quoted are incompatibilities. - both are user errors an .EXE wasn't visible, and a script hadn't been installed (to the right place at least).

Check this one #16731

**Comment by superbonaci** [comment]
> Pretty sure that depends on being on .NET Framework 4. You need to ask the person whose code you're cutting and pasting if they have a version for .NET 6.

Can't we specify PowerShell to use .NET Framework version 4, or to launch older versions of PowerShell?

**Comment by vexx32** [comment]
Keeping compatibility with PowerShell scripts themselves as much as we can has been a priority, yes. However, you're talking about a .NET Core / .NET application being compatible with .NET Framework 4, which has been out of support for many years now (unless you mean 4.8, which I believe is the only version of 4.x still in support?). 

.NET Framework applications and libraries are by and large simply not compatible with newer .NET versions; that's not something PowerShell can do anything about. If you want to run something under Windows PowerShell you can do so, and if you're shipping a module you can specify in the manifest that it only supports specific PowerShell versions or platforms.

If the module manifest does not specify supporting the Core edition of PowerShell, pwsh will typically attempt to load the module in 'windows compatibility mode', where it will actually load the module in Windows PowerShell and communicate between the processes to allow you to use it. This isn't perfect, but it's the closest you'll get without just running Windows PowerShell.

**Comment by superbonaci** [comment]
Also the shortcut should be renamed, from `PowerShell 7 (x64).lnk` to `PowerShell Core 7 (x64).lnk`. Same for 32 bits version and other architectures.

**Comment by superbonaci** [comment]
If I must read documents it means it's not intuitive. Hence I request it's renamed to Ultra or whatever, but not just PowerShell.

**Comment by ImportTaste** [comment]
I agree with the sentiment that dropping 'Core' from the title has made it confusing to properly distinguish between the two, but calling the project 'UltraShell' is just silly.
